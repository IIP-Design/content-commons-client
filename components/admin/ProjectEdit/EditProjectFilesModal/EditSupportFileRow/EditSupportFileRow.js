import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import LanguageDropdown from 'components/admin/dropdowns/LanguageDropdown/LanguageDropdown';
import FileRemoveReplaceButtonGroup from 'components/admin/FileRemoveReplaceButtonGroup/FileRemoveReplaceButtonGroup';
import { truncateAndReplaceStr } from 'lib/utils';
import VisuallyHidden from 'components/VisuallyHidden/VisuallyHidden';
import './EditSupportFileRow.scss';

// Optimize re-renders as component could potentially have many rows
const areEqual = ( prevProps, nextProps ) => {
  const nextFile = nextProps.file;

  //  check for prop differences
  const entries = Object.entries( prevProps.file );
  const same = entries.every( ( [prop, value] ) => {
    if ( prop === 'id' ) {
      return true;
    }

    return nextFile[prop] === value;
  } );

  return same;
};

const EditSupportFileRow = props => {
  const {
    file: { id, name, language }, update, removeFile, accept,
  } = props;

  // extract file type, i.e. get 'image' from the incoming type 'image/*' for example
  // item in first index shows the first capturing group in regex
  // let fileType = /(\w+)\/(\w+)/.exec( type );
  // fileType = ( fileType ) ? fileType[1] : '';
  const filename = ( name && name.length > 25 ) ? truncateAndReplaceStr( name, 25, 8 ) : name;

  return (
    <Grid.Row>

      { /* Filename */ }
      <Grid.Column width={ 8 } className="column">
        <div className="filename">
          { filename !== name
            ? (
              <Fragment>
                <span tooltip={ name }>{ filename }</span>
                <VisuallyHidden el="span">{ name }</VisuallyHidden>
              </Fragment>
            )
            : <span>{ name }</span> }
        </div>
      </Grid.Column>

      { /* Language */ }
      <Grid.Column width={ 6 }>
        <LanguageDropdown id={ id } value={ language } onChange={ update } required />
      </Grid.Column>

      { /* Actions */ }
      <Grid.Column width={ 2 } style={ { paddingLeft: 0 } }>
        <FileRemoveReplaceButtonGroup
          onRemove={ () => { removeFile( id, name ); } }
          accept={ accept }
        />

      </Grid.Column>
    </Grid.Row>
  );
};


EditSupportFileRow.propTypes = {
  file: PropTypes.shape( {
    id: PropTypes.string,
    name: PropTypes.string,
    language: PropTypes.string,
  } ),
  accept: PropTypes.string,
  update: PropTypes.func,
  removeFile: PropTypes.func,
};


export default React.memo( EditSupportFileRow, areEqual );
