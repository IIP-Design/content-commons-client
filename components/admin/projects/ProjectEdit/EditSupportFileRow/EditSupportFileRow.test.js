import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import EditSupportFileRow, {
  DELETE_SUPPORT_FILE_MUTATION,
  UPDATE_SUPPORT_FILE_USE_MUTATION,
  UPDATE_SUPPORT_FILE_LANGUAGE_MUTATION
} from './EditSupportFileRow';

const props = {
  file: {
    id: '8e7eu',
    filename: 'madeinamerica_english.jpg',
    filetype: 'jpg',
    use: {
      id: 'eisu',
      name: 'Thumbnail/Cover Image'
    },
    language: {
      id: 'ewi237',
      displayName: 'English'
    }
  },
  fileExtensions: ['.jpg', '.mp3']
};

const mocks = [
  {
    request: {
      query: UPDATE_SUPPORT_FILE_LANGUAGE_MUTATION,
      variables: {
        data: {
          language: { connect: { id: 'fr143' } }
        },
        where: { id: props.file.id }
      }
    },
    result: {
      data: {
        updateSupportFile: {
          id: props.file.id,
          filename: 'madeinamerica_french.jpg',
          filetype: 'jpg',
          language: {
            id: 'fr143',
            displayName: 'French'
          }
        }
      }
    }
  },
  {
    request: {
      query: UPDATE_SUPPORT_FILE_USE_MUTATION,
      variables: {
        data: {
          use: { connect: { id: 'em92' } }
        },
        where: { id: props.file.id }
      }
    },
    result: {
      data: {
        updateSupportFile: {
          id: props.file.id,
          use: {
            id: 'em92',
            name: 'Email Graphic'
          }
        }
      }
    }
  },
  {
    request: {
      query: DELETE_SUPPORT_FILE_MUTATION,
      variables: { id: props.file.id },
    },
    result: {
      data: {
        deleteSupportFile: {
          id: props.file.id,
          filename: props.file.filename
        }
      }
    }
  }
];

/**
  * Wrap in `<table><tbody></tbody></table>` to prevent
  * this annoying terminal console warning:
  *
  * `Warning: validateDOMNesting(...): <tr> cannot appear as a child of <div>.`
  *
  */
const Component = (
  <MockedProvider mocks={ mocks } addTypename={ false }>
    <table>
      <tbody>
        <EditSupportFileRow { ...props } />
      </tbody>
    </table>
  </MockedProvider>
);

const resizeWindow = ( x, y ) => {
  window.innerWidth = x;
  window.innerHeight = y;
  window.dispatchEvent( new Event( 'resize' ) );
};

describe( '<EditSupportFileRow />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );

    expect( row.exists() ).toEqual( true );
    expect( toJSON( row ) ).toMatchSnapshot();
  } );

  it( '`componentDidMount` sets `_isMounted`', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    inst.componentDidMount();

    expect( inst._isMounted ).toEqual( true );
  } );

  it( '`componentWillUnmount` sets `_isMounted` to false', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    inst.componentWillUnmount();

    expect( inst._isMounted ).toEqual( false );
  } );

  it( 'calls `debounceResize` when window is resized', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    const spy = jest.spyOn( inst, 'debounceResize' );

    inst.componentDidMount();

    const map = {};
    window.addEventListener = jest.fn( ( event, cb ) => {
      map[event] = cb;
    } );

    resizeWindow( 900, 600 );
    expect( spy ).toHaveBeenCalled();
  } );

  it( '`componentWillUnmount` removes the `resize` event listener', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    const map = {};
    window.removeEventListener = jest.fn( ( event, cb ) => {
      map[event] = cb;
    } );

    inst.componentWillUnmount();

    expect( window.removeEventListener )
      .toHaveBeenCalledWith( 'resize', inst.debounceResize );
  } );

  it( '`getShortFileName` returns an ellipsis shortened file name', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    const longName = props.file.filename; // madeinamerica_english.jpg
    const index = 8;
    const shortName = inst
      .getShortFileName( longName, index )
      .props
      .children;

    expect( shortName ).toHaveLength( 3 );
    expect( shortName[0] ).toHaveLength( 8 );
    expect( shortName[1] ).toHaveLength( 1 );
    expect( shortName[2] ).toHaveLength( 8 );
    expect( shortName.join( '' ) ).toEqual( 'madeinamâ€¦lish.jpg' );
  } );

  it( '`getProportionalNumber` returns a number that is proportional to a reference', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    const reference = 500;
    const proportion = inst.ITEM_NAME_PROPORTION;
    const num = inst
      .getProportionalNumber( reference, proportion );

    expect( num ).toEqual( 425 );
  } );

  it( '`isLongName` returns a boolean whether given number is greater or equal to another', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    let itemWidth = 290;
    const reference = 500;
    const proportion = inst.ITEM_NAME_PROPORTION;

    const isLongName = () => (
      inst.isLongName( itemWidth, reference, proportion )
    );

    expect( isLongName() ).toEqual( false );
    itemWidth = 450;
    expect( isLongName() ).toEqual( true );
  } );

  it( '`setRefWidth` and `resetWidths` sets ref widths in state', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    const div = row.find( 'div.file-name' );
    const span = row.find( 'span.file-name-wrap' );

    const nodes = [div, span];
    const names = ['cell', 'fileName'];

    div.offsetWidth = 200;
    span.offsetWidth = 170;

    nodes.forEach( ( n, i ) => {
      expect( row.state( `${names[i]}Width` ) )
        .toEqual( 0 );

      inst.setRefWidth( n, names[i] );
      expect( row.state( `${names[i]}Width` ) )
        .toEqual( n.offsetWidth );

      inst.resetWidths();
      expect( row.state( `${names[i]}Width` ) )
        .toEqual( 0 );
    } );
  } );

  it( '`resetWidths` does not reset ref widths in state if component is unmounted', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();

    const div = row.find( 'div.file-name' );
    const span = row.find( 'span.file-name-wrap' );

    const nodes = [div, span];
    const names = ['cell', 'fileName'];

    div.offsetWidth = 200;
    span.offsetWidth = 170;

    inst.componentDidMount();
    const isMounted = () => inst._isMounted;

    expect( isMounted() ).toEqual( true );

    nodes.forEach( ( n, i ) => {
      expect( row.state( `${names[i]}Width` ) )
        .toEqual( 0 );

      inst.setRefWidth( n, names[i] );
      expect( row.state( `${names[i]}Width` ) )
        .toEqual( n.offsetWidth );
    } );

    inst.componentWillUnmount();
    expect( isMounted() ).toEqual( false );

    inst.resetWidths();

    nodes.forEach( ( n, i ) => {
      expect( row.state( `${names[i]}Width` ) )
        .toEqual( n.offsetWidth );
    } );
  } );

  it( 'renders `setReplaceFileRef`', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();
    const input = wrapper.find( 'input#upload-file--single' );

    expect( input.exists() ).toEqual( true );
    expect( inst.setReplaceFileRef ).toBeTruthy();
  } );

  it( 'sets correct `acceptedTypes` for the file upload `input`', () => {
    const newProps = { ...props, ...{ fileExtensions: ['.srt'] } };
    const wrapper = mount(
      <MockedProvider mocks={ mocks } addTypename={ false }>
        <table>
          <tbody>
            <EditSupportFileRow { ...newProps } />
          </tbody>
        </table>
      </MockedProvider>
    );

    const input = () => wrapper.find( 'input#upload-file--single' );

    expect( input().prop( 'accept' ) ).toEqual( '.srt' );
  } );

  it( '`handleChange` sets `fileLanguageId` in state', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = () => row.instance();

    const e = {};
    const selection = { value: 'fr143', name: 'fileLanguageId' };

    expect( inst().state.fileLanguageId ).toEqual( 'ewi237' );

    inst().handleChange( e, selection );

    inst().forceUpdate();

    expect( inst().state.fileLanguageId ).toEqual( 'fr143' );
  } );

  it( '`handleChange` sets `fileUse` in state', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = () => row.instance();

    const e = {};
    const selection = { value: 'em92', name: 'fileUse' };

    expect( inst().state.fileUse ).toEqual( 'eisu' );

    inst().handleChange( e, selection );

    expect( inst().state.fileUse ).toEqual( 'em92' );
  } );

  it( 'Popup is rendered if file name is long', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const popup = () => (
      wrapper.find( `Popup[content="${props.file.filename}"]` )
    );

    /**
     * file name is long if `isLongName` is `true`, i.e.,
     * fileNameWidth/cellWidth >= 0.85 (ITEM_NAME_PROPORTION)
     */
    row.setState( { cellWidth: 400, fileNameWidth: 350 } );
    expect( popup().exists() ).toEqual( true );

    row.setState( { cellWidth: 400, fileNameWidth: 300 } );
    expect( popup().exists() ).toEqual( false );
  } );

  it( '`fileName` only (i.e., no Popup) is rendered if file name is not long', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const span = wrapper.find( 'span.file-name-wrap' );

    row.setState( { cellWidth: 400, fileNameWidth: 300 } );
    const popup = wrapper.find( `Popup[content="${props.file.filename}"]` );

    expect( popup.exists() ).toEqual( false );
    expect( span.text() ).toEqual( props.file.filename );
  } );

  it( 'clicking Delete icon calls `handleDeleteFile`', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();
    const deleteBtn = row.find( 'Button.delete' );
    const spy = jest.spyOn( inst, 'handleDeleteFile' );

    inst.forceUpdate();
    deleteBtn.simulate( 'click' );

    expect( spy ).toHaveBeenCalled();
  } );

  it( 'clicking Replace icon calls `handleReplaceFile`', () => {
    const wrapper = mount( Component );
    const row = wrapper.find( 'EditSupportFileRow' );
    const inst = row.instance();
    const replaceBtn = row.find( 'Button.replace' );
    const spy = jest.spyOn( inst, 'handleReplaceFile' );
    inst.addReplaceFileRef = {
      click: jest.fn()
    };

    inst.forceUpdate();
    replaceBtn.simulate( 'click' );

    expect( spy ).toHaveBeenCalled();
    expect( inst.addReplaceFileRef.click ).toHaveBeenCalled();
  } );
} );
