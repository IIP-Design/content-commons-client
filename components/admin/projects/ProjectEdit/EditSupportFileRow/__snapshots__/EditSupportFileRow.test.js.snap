// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditSupportFileRow /> renders without crashing 1`] = `
<EditSupportFileRow
  deleteFile={[Function]}
  file={
    Object {
      "filename": "madeinamerica_english.jpg",
      "filetype": "jpg",
      "id": "8e7eu",
      "language": Object {
        "displayName": "English",
        "id": "ewi237",
      },
      "use": Object {
        "id": "eisu",
        "name": "Thumbnail/Cover Image",
      },
    }
  }
  fileExtensions={
    Array [
      ".jpg",
      ".mp3",
    ]
  }
  updateFileUse={[Function]}
  updateLanguage={[Function]}
>
  <TableRow
    as="tr"
    cellAs="td"
  >
    <tr
      className=""
    >
      <TableCell
        as="td"
      >
        <td
          className=""
        >
          <div
            className="file-name"
          >
            <VisuallyHidden>
              <div
                className="hide-visually"
              >
                madeinamerica_english.jpg
              </div>
            </VisuallyHidden>
            <span
              aria-hidden={true}
              className="file-name-wrap hasEllipsis"
            >
              <Popup
                content="madeinamerica_english.jpg"
                inverted={true}
                keepInViewPort={true}
                on={
                  Array [
                    "hover",
                    "click",
                    "focus",
                  ]
                }
                position="top left"
                size="mini"
                style={
                  Object {
                    "msWordBreak": "break-all",
                    "overflowWrap": "break-word",
                    "wordBreak": "break-word",
                    "wordWrap": "break-word",
                  }
                }
                trigger={
                  <span>
                    <Focusable
                      order={0}
                    >
                      <React.Fragment>
                        
                        …
                        madeinamerica_english.jpg
                      </React.Fragment>
                    </Focusable>
                  </span>
                }
              >
                <Portal
                  closeOnDocumentClick={true}
                  closeOnEscape={true}
                  closeOnTriggerBlur={true}
                  closeOnTriggerClick={true}
                  closeOnTriggerMouseLeave={true}
                  eventPool="default"
                  mouseEnterDelay={50}
                  mouseLeaveDelay={70}
                  onClose={[Function]}
                  onMount={[Function]}
                  onOpen={[Function]}
                  onUnmount={[Function]}
                  openOnTriggerClick={true}
                  openOnTriggerFocus={true}
                  openOnTriggerMouseEnter={true}
                  trigger={
                    <span>
                      <Focusable
                        order={0}
                      >
                        <React.Fragment>
                          
                          …
                          madeinamerica_english.jpg
                        </React.Fragment>
                      </Focusable>
                    </span>
                  }
                  triggerRef={[Function]}
                >
                  <Ref
                    innerRef={[Function]}
                  >
                    <span
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <Focusable
                        order={0}
                      >
                        <span
                          className="focusable"
                          tabIndex={0}
                        >
                          …
                          madeinamerica_english.jpg
                        </span>
                      </Focusable>
                    </span>
                  </Ref>
                </Portal>
              </Popup>
            </span>
          </div>
        </td>
      </TableCell>
      <TableCell
        as="td"
      >
        <td
          className=""
        >
          <VisuallyHidden>
            <div
              className="hide-visually"
            >
              <label
                htmlFor="file-8e7eu"
              >
                madeinamerica_english.jpg language
              </label>
            </div>
          </VisuallyHidden>
          <Apollo(DropdownLanguage)
            fluid={true}
            id="file-8e7eu"
            name="fileLanguageId"
            onChange={[Function]}
            required={true}
            value="ewi237"
          >
            <Query
              displayName="Apollo(DropdownLanguage)"
              query={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "Languages",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orderBy",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "orderBy",
                                  },
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "languages",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": Object {
                                    "kind": "Name",
                                    "value": "value",
                                  },
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": Object {
                                    "kind": "Name",
                                    "value": "text",
                                  },
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "displayName",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [
                        Object {
                          "defaultValue": undefined,
                          "kind": "VariableDefinition",
                          "type": Object {
                            "kind": "NamedType",
                            "name": Object {
                              "kind": "Name",
                              "value": "LanguageOrderByInput",
                            },
                          },
                          "variable": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "orderBy",
                            },
                          },
                        },
                      ],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 138,
                    "start": 0,
                  },
                }
              }
              skip={false}
              variables={
                Object {
                  "orderBy": "displayName_ASC",
                }
              }
              warnUnhandledError={true}
            >
              <DropdownLanguage
                data={
                  Object {
                    "error": undefined,
                    "fetchMore": [Function],
                    "loading": true,
                    "networkStatus": 1,
                    "refetch": [Function],
                    "startPolling": [Function],
                    "stopPolling": [Function],
                    "subscribeToMore": [Function],
                    "updateQuery": [Function],
                    "variables": Object {
                      "orderBy": "displayName_ASC",
                    },
                  }
                }
                fluid={true}
                id="file-8e7eu"
                name="fileLanguageId"
                onChange={[Function]}
                required={true}
                value="ewi237"
              >
                Loading language...
              </DropdownLanguage>
            </Query>
          </Apollo(DropdownLanguage)>
        </td>
      </TableCell>
      <TableCell
        as="td"
      >
        <td
          className=""
        >
          <VisuallyHidden>
            <div
              className="hide-visually"
            >
              <label
                htmlFor="use-8e7eu"
              >
                madeinamerica_english.jpg use
              </label>
            </div>
          </VisuallyHidden>
          <Apollo(DropdownSupportFileUse)
            fluid={true}
            id="use-8e7eu"
            name="fileUse"
            onChange={[Function]}
            required={true}
            value="eisu"
          >
            <Query
              displayName="Apollo(DropdownSupportFileUse)"
              query={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "SupportFileUse",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": Object {
                              "kind": "Name",
                              "value": "uses",
                            },
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orderBy",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "orderBy",
                                  },
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "supportFileUses",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": Object {
                                    "kind": "Name",
                                    "value": "value",
                                  },
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": Object {
                                    "kind": "Name",
                                    "value": "text",
                                  },
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "name",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [
                        Object {
                          "defaultValue": undefined,
                          "kind": "VariableDefinition",
                          "type": Object {
                            "kind": "NamedType",
                            "name": Object {
                              "kind": "Name",
                              "value": "SupportFileUseOrderByInput",
                            },
                          },
                          "variable": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "orderBy",
                            },
                          },
                        },
                      ],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 154,
                    "start": 0,
                  },
                }
              }
              skip={false}
              variables={
                Object {
                  "orderBy": "name_ASC",
                }
              }
              warnUnhandledError={true}
            >
              <DropdownSupportFileUse
                data={
                  Object {
                    "error": undefined,
                    "fetchMore": [Function],
                    "loading": true,
                    "networkStatus": 1,
                    "refetch": [Function],
                    "startPolling": [Function],
                    "stopPolling": [Function],
                    "subscribeToMore": [Function],
                    "updateQuery": [Function],
                    "variables": Object {
                      "orderBy": "name_ASC",
                    },
                  }
                }
                fluid={true}
                id="use-8e7eu"
                name="fileUse"
                onChange={[Function]}
                required={true}
                value="eisu"
              >
                Loading file use...
              </DropdownSupportFileUse>
            </Query>
          </Apollo(DropdownSupportFileUse)>
        </td>
      </TableCell>
      <TableCell
        as="td"
      >
        <td
          className=""
        >
          <ButtonGroup
            basic={true}
            size="large"
          >
            <div
              className="ui large basic buttons"
            >
              <Popup
                content="Replace"
                inverted={true}
                keepInViewPort={true}
                on={
                  Array [
                    "hover",
                    "click",
                    "focus",
                  ]
                }
                position="top left"
                size="mini"
                trigger={
                  <Button
                    aria-label="replace file"
                    as="button"
                    basic={true}
                    className="replace"
                    icon="refresh"
                    onClick={[Function]}
                  />
                }
              >
                <Portal
                  closeOnDocumentClick={true}
                  closeOnEscape={true}
                  closeOnTriggerBlur={true}
                  closeOnTriggerClick={true}
                  closeOnTriggerMouseLeave={true}
                  eventPool="default"
                  mouseEnterDelay={50}
                  mouseLeaveDelay={70}
                  onClose={[Function]}
                  onMount={[Function]}
                  onOpen={[Function]}
                  onUnmount={[Function]}
                  openOnTriggerClick={true}
                  openOnTriggerFocus={true}
                  openOnTriggerMouseEnter={true}
                  trigger={
                    <Button
                      aria-label="replace file"
                      as="button"
                      basic={true}
                      className="replace"
                      icon="refresh"
                      onClick={[Function]}
                    />
                  }
                  triggerRef={[Function]}
                >
                  <Ref
                    innerRef={[Function]}
                  >
                    <Button
                      aria-label="replace file"
                      as="button"
                      basic={true}
                      className="replace"
                      icon="refresh"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <button
                        aria-label="replace file"
                        className="ui basic icon button replace"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <Icon
                          as="i"
                          name="refresh"
                        >
                          <i
                            aria-hidden="true"
                            className="refresh icon"
                          />
                        </Icon>
                      </button>
                    </Button>
                  </Ref>
                </Portal>
              </Popup>
              <VisuallyHidden>
                <div
                  className="hide-visually"
                >
                  <label
                    htmlFor="upload-file--single"
                  >
                    upload file
                  </label>
                  <input
                    accept=""
                    id="upload-file--single"
                    tabIndex={-1}
                    type="file"
                  />
                </div>
              </VisuallyHidden>
              <Popup
                content="Delete"
                inverted={true}
                keepInViewPort={true}
                on={
                  Array [
                    "hover",
                    "click",
                    "focus",
                  ]
                }
                position="top left"
                size="mini"
                trigger={
                  <Button
                    aria-label="delete file"
                    as="button"
                    basic={true}
                    className="delete"
                    icon="delete"
                    onClick={[Function]}
                  />
                }
              >
                <Portal
                  closeOnDocumentClick={true}
                  closeOnEscape={true}
                  closeOnTriggerBlur={true}
                  closeOnTriggerClick={true}
                  closeOnTriggerMouseLeave={true}
                  eventPool="default"
                  mouseEnterDelay={50}
                  mouseLeaveDelay={70}
                  onClose={[Function]}
                  onMount={[Function]}
                  onOpen={[Function]}
                  onUnmount={[Function]}
                  openOnTriggerClick={true}
                  openOnTriggerFocus={true}
                  openOnTriggerMouseEnter={true}
                  trigger={
                    <Button
                      aria-label="delete file"
                      as="button"
                      basic={true}
                      className="delete"
                      icon="delete"
                      onClick={[Function]}
                    />
                  }
                  triggerRef={[Function]}
                >
                  <Ref
                    innerRef={[Function]}
                  >
                    <Button
                      aria-label="delete file"
                      as="button"
                      basic={true}
                      className="delete"
                      icon="delete"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <button
                        aria-label="delete file"
                        className="ui basic icon button delete"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <Icon
                          as="i"
                          name="delete"
                        >
                          <i
                            aria-hidden="true"
                            className="delete icon"
                          />
                        </Icon>
                      </button>
                    </Button>
                  </Ref>
                </Portal>
              </Popup>
            </div>
          </ButtonGroup>
        </td>
      </TableCell>
    </tr>
  </TableRow>
</EditSupportFileRow>
`;
